tests/pylint/test_enforce_type_hints.py

```diff
@@ -937,3 +940,39 @@ def native_value(self) -> int:
 
     with assert_no_messages(linter):
         type_hint_checker.visit_classdef(class_node)
+
+
+def test_vacuum_entity(linter: UnittestLinter, type_hint_checker: BaseChecker) -> None:
+    """Ensure valid hints are accepted for vacuum entity."""
+    # Set bypass option
+    type_hint_checker.config.ignore_missing_annotations = False
+
+    # Ensure that `dict | list | None` is valid for params
+    class_node = astroid.extract_node(
+        """
+    class Entity():
+        pass
+
+    class ToggleEntity(Entity):
+        pass
+
+    class _BaseVacuum(Entity):
+        pass
+
+    class VacuumEntity(_BaseVacuum, ToggleEntity):
+        pass
+
+    class MyVacuum( #@
+        VacuumEntity
+    ):
+        def send_command(
+            self, command: str, params: dict | list | None = None, **kwargs: Any
+        ) -> None:
```

Comment:
What happens / should happen with `params: dict[str, Any] | list[Any] | None`?

Reply:
I considered adjusting the type hint in the base vacuum entity to that. Maybe ghat would be better?

Reply:
> I considered adjusting the type hint in the base vacuum entity to that. Maybe ghat would be better?

That's probably best. We're far from enabling it everywhere, but a bare `dict` or `list` would trigger `disallow_any_generics`.

Reply:
👎 regex needs more work

Reply:
I have adjusted the regex, should be ok now...